rasa/core/actions/forms.py

```diff
@@ -664,7 +667,25 @@ async def is_done(
         domain: "Domain",
         events_so_far: List[Event],
     ) -> bool:
-        return SlotSet(REQUESTED_SLOT, None) in events_so_far
+        # custom validation functions can decide to terminate the loop early by
+        # setting the requested slot to `None` or setting `ActiveLoop(None)`
+        return next(
+            (
+                event
+                for event in reversed(events_so_far)
+                if isinstance(event, SlotSet) and event.key == REQUESTED_SLOT
+            ),
+            None,
+        ) == SlotSet(REQUESTED_SLOT, None) or next(
+            (
+                event
+                for event in reversed(events_so_far)
+                if isinstance(event, ActiveLoop)
+            ),
+            None,
+        ) == ActiveLoop(
+            None
+        )
```

Comment:
`in` operator doesn't work?

Reply:
`SlotSet(REQUESTED_SLOT, None) in events_so_far` seems to me much more readable

Reply:
I explicitly wanna only check against the last occurrences of these events. I wanna avoid that something earlier returns `ActiveLoop(None)` but something later is trying to override it with `ActiveLoop("dasdasd")` and it doesn't work. It should usually be the same, but I think this is slightly more robust

Reply:
ok, not sure if we should catch multiple settings differently, but for now could you please add this comment to the code, so that in the future we could return to it, if smth doesn't work as expected

Reply:
done, do I get a 👍 then?